actions/helm_chart_e2e.go

```diff
@@ -0,0 +1,130 @@
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	// RepoFlag represents the '-repo' flag
+	RepoFlag = "repo"
+	// RefFlag represents the '-ref' flag (for specifying a SHA, branch or tag)
+	RefFlag = "ref"
+)
+
+// HelmChartE2E is the cli handler for generating a release helm chart for deis-e2e
+func HelmChartE2E(ghClient *github.Client) func(*cli.Context) {
+	return func(c *cli.Context) {
+		chartDir := "workflow-dev-e2e"
+		downloadFiles(ghClient, c, chartDir, []string{"README.md", "Chart.yaml"})
+
+		err := updateFilesWithRelease(chartDir)
+		if err != nil {
+			log.Fatalf("Error updating files with release (%s)", err)
+		}
+	}
+}
+
+func createSubDir(dirName string) {
+	dir := fmt.Sprintf("%s/%s", stagingDirName, dirName)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err := os.Mkdir(dir, os.ModePerm)
+		if err != nil {
+			log.Fatalf("Error creating dir %s (%s)", dir, err)
+		}
+	}
+}
+
+func downloadFiles(ghClient *github.Client, c *cli.Context, rootDir string, fileNames []string) {
+	var opt github.RepositoryContentGetOptions
+	opt.Ref = c.GlobalString(RefFlag)
+
+	createSubDir(rootDir)
+
+	for _, fileName := range fileNames {
+		filePath := fmt.Sprintf("%s/%s", rootDir, fileName)
+		readCloser, err := downloadContents(ghClient, c.GlobalString(OrgFlag), c.GlobalString(RepoFlag), filePath, &opt)
+		if err != nil {
+			log.Fatalf("Error downloading contents of %s (%s)", filePath, err)
+		} else {
+			defer readCloser.Close()
+			contents, err := ioutil.ReadAll(readCloser)
+			if err != nil {
+				log.Fatalf("Error reading contents of %s (%s)", filePath, err)
+			}
+			// create file in stagingDirName and write contents to it
+			currentWorkingDir, err := os.Getwd()
+			if err != nil {
+				log.Fatalf("Error getting current working dir (%s)", err)
+			}
+			newFileName := fmt.Sprintf("%s/%s/%s", currentWorkingDir, stagingDirName, filePath)
+			f, err := os.Create(newFileName)
+			if err != nil {
+				log.Fatalf("Error creating file %s (%s)", newFileName, err)
+			}
+			defer f.Close()
+			_, err = f.Write(contents)
+			if err != nil {
+				log.Fatalf("Error writing contents to file %s (%s)", newFileName, err)
+			}
+			f.Sync()
+		}
+	}
+}
+
+func updateFilesWithRelease(dirToWalk string) error {
+	currentWorkingDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Error getting current working dir (%s)", err)
+	}
+	// walk and replace with "dev" with DEIS_RELEASE_SHORT
+	err = filepath.Walk(fmt.Sprintf("%s/%s/%s", currentWorkingDir, stagingDirName, dirToWalk), makeRelease)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func makeRelease(path string, fi os.FileInfo, err error) error {
+	deisRelease := os.Getenv("DEIS_RELEASE")
+	if deisRelease == "" {
+		log.Fatalf("'DEIS_RELEASE' env var required")
+	}
+	deisReleaseShort := os.Getenv("DEIS_RELEASE_SHORT")
+	if deisReleaseShort == "" {
+		log.Fatalf("'DEIS_RELEASE_SHORT' env var required")
+	}
```

Comment:
@arschles added description as requested -- also note that `helm-chart` action has been renamed to more appropriate `helm-stage`